imapserver/imapmemserver/session.go

```diff
@@ -17,13 +17,35 @@ type (
 type UserSession struct {
 	*user    // immutable
 	*mailbox // may be nil
+
+	// appendLimit is the maximum size in bytes that can be uploaded to this server
+	// in an APPEND command
+	appendLimit uint32
+
+	// discloseLimit indicates whether the append limit should be advertised in the
+	// CAPABILITY response
+	discloseLimit bool
 }
 
 var _ imapserver.SessionIMAP4rev2 = (*UserSession)(nil)
+var _ imapserver.SessionAppendLimit = (*UserSession)(nil)
 
 // NewUserSession creates a new user session.
 func NewUserSession(user *User) *UserSession {
-	return &UserSession{user: user}
+	return &UserSession{
+		user:          user,
+		appendLimit:   104857600, // 100 MiB default
+		discloseLimit: true,      // By default, disclose the limit in CAPABILITY
+	}
+}
+
+// NewUserSessionWithAppendLimit creates a new user session with a custom append limit.
+func NewUserSessionWithAppendLimit(user *User, appendLimit uint32, discloseLimit bool) *UserSession {
```

Comment:
Could we keep this simple and only have a single global append limit in imapmemserver?